lib/puppet/provider/oneview_enclosure_group/c7000.rb

```diff
@@ -22,36 +22,27 @@
   confine true: login[:hardware_variant] == 'C7000'
 
   mk_resource_methods
-  def exists?
-    @data = enclosure_group_parse(data_parse)
-    empty_data_check
-    !@resource_type.find_by(@client, @data).empty?
-  end
-
-  def create
-    return true if resource_update
-    @resource_type.new(@client, enclosure_group_parse(@data)).create
-  end
 
   def get_script
     Puppet.notice("Enclosure Group's current script: \n#{get_single_resource_instance.get_script}\n")
   end
 
   def set_script
-    script = @data.delete('script') if @data['script']
+    script = @data.delete('script')
     raise("\nThe 'script' field is required in data hash to run the set_script action.") unless script
```

Comment:
script = @data.delete('script') || raise("\nThe 'script' field is required in data hash to run the set_script action.")